src/seismometer/controls/thresholds.py

```diff
@@ -0,0 +1,154 @@
+        self._disabled = disabled
+        for slider in self.sliders.values():
+            slider.disabled = disabled
+
+    def _on_slider_change(self, change=None):
+        """Slider has changed, update the value tuple"""
+        if self.value_update_in_progress:
+            return
+        if change:
+            if (slider := change["owner"]) in self.sliders.values():
+                self.value[slider.description] = slider.value
+
+    def _on_value_change(self, change=None):
+        """Bubble up changes to sliders"""
+        self.value_update_in_progress = True
+        if change:
+            if val := change["new"]:
+                for key, value in val.items():
+                    self.sliders[key].value = value
+        self.value_update_in_progress = False
+
+
+class MonotonicProbabilitySliderListWidget(ProbabilitySliderListWidget):
+    """
+    Vertical list of sliders, bounded between 0 and 1 with a step of 0.01.
+    Monotonicity is maintained between the sliders so they are always ascending or descending in value.
+    """
+
+    def __init__(self, names: tuple[str], value: tuple[int] = None, ascending: bool = True):
+        """A vertical list of sliders, bounded between 0 and 1 with a step of 0.01.
+
+        Parameters
+        ----------
+        names : tuple[str]
+            Slider names
+        value : Optional[tuple[str]], optional
+            Slider start values, by default None, starts all sliders at zero.
+        ascending : bool, optional = True
+            Forces sliders to be ascending, else decreasing.
+            If initial values are not sorted, raise an ValueError.
+        """
+        if len(names) > 6:
+            raise ValueError("MonotonicProbabilitySliderListWidget only supports up to 6 sliders")
+
+        super().__init__(names, value)
+
+        if tuple(sorted(self.value.values(), reverse=not ascending)) != tuple(self.value.values()):
+            direction = "ascending" if ascending else "descending"
+            raise ValueError(f"Initial values are not sorted, expected {direction}")
+
+        self.ascending = ascending
+        for index, sub_slider in enumerate(self.sliders.values()):
+            if ascending:
+                # line them up so that last slider is the first alert color
+                sub_slider.style.handle_color = alert_colors[(len(self.value) - index - 1) % len(alert_colors)]
+            else:
+                sub_slider.style.handle_color = alert_colors[index % len(alert_colors)]
+
+    def _on_slider_change(self, change=None):
+        """Slider has changed, update the value tuple, making sure values are increating"""
+        sliders = list(self.sliders.values())
+        if self.value_update_in_progress:
+            return
+        try:
+            slider_index = sliders.index(change["owner"])
+        except ValueError:  # Only the sliders can change values
+            return
+        new = change["new"]
+        old = change["old"]
```

Comment:
If change is default None, would this fail?

If change["owner"] exists but change{'new"] nad change["old"] doesn't, would this fail?

Reply:
If owner exists, we will have a new and old value.